### ISSUE TITLE
getEquivalentVariables needs to check that the term is in fact a variable in the PC

### ISSUE THREAD

**Issue by jennalwise** [gradual-verification/silicon-gv#23]
This program (https://github.com/gradual-verification/gvc0/blob/master/src/test/resources/bugs/conditional_permissions.c0) is failing with a `// ERROR: java.lang.RuntimeException: Match error in resolveType!` error. The error happens when translating the branch condition ` (== input@24@01.A$b@31@01 0)` when creating a run-time check for the consumption of `acc(input->c)` at line 17. The consumption ensures the precondition of `requirePermissions(input)` holds at line 17. `input@24@01.A$b@31@01` does not have a corresponding field chunk in the Heap and OH no matter what symbolic state is used to translate it. So, `variableResolverHelper` returns `None` and so `PathConditions.getEquivalentVariables` is called on `input@24@01.A$b@31@01` and this is where the problem lies. This function looks for aliased variables to the one in question in the path condition, but it does not check that the aliased thing in the path condition is actually a variable. So, since `(== input@24@01.A$b@31@01 0)` is in the path condition 0 (an Int Lit) is being returned, which is not a variable, but is being resolved like one. Hence, the resolveType match error.
Same problem occurs in https://github.com/gradual-verification/gvc0/blob/master/src/test/resources/bugs/loop_something.c0 for the negated while loop condition: `n->next == NULL` when the postcondition of append is being consumed.

**Comment by hgouni** [comment]
This has been fixed by 0deb0b9a, but should be kept open for now, while we figure out what needs to be done for translating branches.

**Comment by jennalwise** [comment]
After fixing issue https://github.com/gradual-verification/silicon-gv/issues/29, I discovered that the `getEquivalentVariables` fix is not quite right. We need the verifier to look for aliased variables to the given term, we do not want to make sure the term that we have is a variable, ie. the `getEquivalentVariables` fix is now too restrictive. It is only looking for aliased symbolic variables on both sides. We want the term we are given to be whatever, but the term that it aliases too must be a variable. I fixed this as well in this commit: https://github.com/gradual-verification/silicon-gv/commit/b33d60aa38a33203316fcf0baf270c03dce58bec .
